Improve grammar, spelling and structure of HTTP-Header Format and processing model

danielkhan opened the pull request:
This PR contains a set of changes to the HTTP Header Format specification and the processing model document.
Except replacing the 512byte limit with the 5 list item limit, there should be no semantic changes included.

It contains
- Spelling fixes
- Grammatical fixes
- Changes that add more consistency in our wording
- Some restructuring to make the overall flow more coherent

This PR is meant to be a base for further discussions and should not be considered as final. Please review thoroughly, if the changes make sense and that no semantic changes gave slipped in.

danielkhan committed:
Grammar, Spelling and wording

- Fixed grammar, spelling and edited wording for clarity
- Formatted pending decisions
- Added issue links

danielkhan committed:
Moved IF conditions to bullets instead of ordered lists

danielkhan committed:
Auto-fixed spelling and grammar

danielkhan committed:
Edited and restructured for more clarity

danielkhan committed:
Edited example in intro

danielkhan committed:
Intro example - titles

danielkhan committed:
Edited and restructured for more consistency

Rephrased the intro example.

danielkhan committed:
Markdown linting

danielkhan committed:
ignore vscode files

danielkhan committed:
Removing vscode specific file

danielkhan committed:
removed vscode folder

danielkhan committed:
Merge branch 'cr-edit-http-header-format' of github.com:danielkhan/distributed-tracing into cr-edit-http-header-format

danielkhan committed:
Fix line lengths

AloisReitbauer commented on spec/20-HTTP_HEADER_FORMAT.md:
```diff
@@ -73,178 +76,198 @@ value           = version "-" version-format
-value. There are systems that make random sampling decisions based on the value
-of `trace-id`. So to increase interoperability it is recommended to keep the
-random part on the right side of `trace-id` value.
+**Example:** `4bf92f3577b34da6a3ce929d0e0e4736`.
```
use uppercase for hexdig

AloisReitbauer commented on spec/20-HTTP_HEADER_FORMAT.md:
```diff
@@ -73,178 +76,198 @@ value           = version "-" version-format
-example, `4bf92f3577b34da6a3ce929d0e0e4736`. All bytes `0` are considered
-invalid.
+Represents the ID of the whole trace forest as 16-byte array and is used to
+uniquely identify a distributed trace throughout all participating systems.
```
add detailed semantics

AloisReitbauer commented on spec/20-HTTP_HEADER_FORMAT.md:
```diff
@@ -73,178 +76,198 @@ value           = version "-" version-format
-even though a system may operate with a shorter `trace-id` for distributed trace
-reporting - full `trace-id` should be propagated to conform to the
-specification.
+> **Note:** Many algorithms of unique ID generation are based on some constant
```
for

AloisReitbauer commented on spec/20-HTTP_HEADER_FORMAT.md:
```diff
@@ -73,178 +76,198 @@ value           = version "-" version-format
-reporting - full `trace-id` should be propagated to conform to the
-specification.
+> **Note:** Many algorithms of unique ID generation are based on some constant
+> part (time or host-based) and a random value. Some systems make random
```
Isn't this a too opinionated way to talk about randomness?

danielkhan commented on spec/20-HTTP_HEADER_FORMAT.md:
```diff
@@ -73,178 +76,198 @@ value           = version "-" version-format
+* A `trace-id` with all bytes set to `0` is not allowed.
+* Implementations SHOULD generate globally unique values.
+* Implementations HAVE TO ignore the `traceparent` if the `trace-id` does not
+* comply with this specification.
```
fix bullets

AloisReitbauer commented on spec/20-HTTP_HEADER_FORMAT.md:
```diff
@@ -73,178 +76,198 @@ value           = version "-" version-format
-2. Bug in caller
-3. Different load between caller service and callee service might force callee
-   to down sample.
+**Note:** `parent-id` is sometimes also referred to as `span-id` as some
```
remove

AloisReitbauer commented on spec/20-HTTP_HEADER_FORMAT.md:
```diff
@@ -73,178 +76,198 @@ value           = version "-" version-format
-## Versioning of `traceparent`
+### Versioning of `traceparent`
+
+> **FIXME - what does this mean?** Implementation is opinionated about future version of the specification.
```
remove sentence

AloisReitbauer commented on spec/20-HTTP_HEADER_FORMAT.md:
```diff
@@ -360,70 +362,94 @@ nblk-chr = %x21-2B / %x2D-3C / %x3E-7E
+  the system SHOULD start dropping items starting at the right side (end) of the
+  list. When propagating `tracestate` with the excessive length, the assumption
+  SHOULD be that the receiver will trim the list to the length of five.
+* Libraries and platforms MUST accept empty `tracestate` headers, but SHOULD
```
TMI

AloisReitbauer commented on spec/20-HTTP_HEADER_FORMAT.md:
```diff
@@ -467,13 +493,13 @@ the beginning of the list.
 Here is the list of allowed mutations:
```
wording

danielkhan committed:
Improved wording wrt byte arrays / hex

danielkhan committed:
Fixed formatting

danielkhan committed:
Removed generic tracer paragraph

This seems to be an implementation detail that should likely not be part of the spec.

danielkhan committed:
Removed paragraph about reuse of trace ID for correlation.

danielkhan committed:
Added 'field' when mentioning HTTP-header

danielkhan committed:
Removed redundant explanation of OWS

danielkhan committed:
Reverted the removed paragraphs to allow further discussion

danielkhan commented on spec/20-HTTP_HEADER_FORMAT.md:
````diff
@@ -1,30 +1,41 @@
 tracestate: rojo=00-0af7651916cd43dd8448eb211c80319c-00f067aa0ba902b7-01,congo=BleGNlZWRzIHRohbCBwbGVhc3VyZS4
 ```
 
+> **FIXME - the generic tracing system is defined in the rationale doc which is
````
Clarify, if this information isn't too much information in this context

danielkhan committed:
Changed wording from list to ordered set

danielkhan committed:
Changed wording to set

danielkhan committed:
Removed 'opnionated' remark

danielkhan committed:
changed wording to field value
